### PR TITLE
Migrate specs update to updatecli

### PR DESCRIPTION
**Pull request by reakaleek**
## What

Migrate https://github.com/elastic/apm/blob/main/.ci/Jenkinsfile to the consumer using updatecli and GH Actions workflows.

## Why

Jenkins 2 GH Actions Migration

## How to test

```shell
export GITHUB_TOKEN="<your-personal-access-token>"
export GIT_USER="FooBar"
export GIT_EMAIL="foo@bar.com"
updatecli --config .ci/updatecli.d/update-json-specs.yml diff
```

### CODE REVIEW

File: .github/workflows/updatecli.yml [hunk 1]
```diff
@@ -14,16 +14,15 @@ jobs:
...
         with:
           vaultUrl: ${{ secrets.VAULT_ADDR }}
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
-          pipeline: ./.ci/update-specs.yml
+          pipeline: .ci/updatecli.d
       - if: failure()
         uses: elastic/apm-pipeline-library/.github/actions/notify-build-status@current
         with:
+          status: ${{ steps.check.outputs.status }}
```
Comment (v1v) [inline .github/workflows/updatecli.yml:25]: leftover?

```suggestion
```

Comment (reakaleek) [reply .github/workflows/updatecli.yml:25]: Thank you for your eagle eyes. 🦅

Comment (reakaleek) [reply .github/workflows/updatecli.yml]: https://github.com/elastic/apm-agent-nodejs/pull/3197/commits/2a3c66a84b413aee36c6b24a3fe1729eea198c14

**Review by v1v — APPROVED** [review]

File: .ci/updatecli.d/update-json-specs.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/cgroup_parsing.json
+  service_resource_inference.json:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/service_resource_inference.json
+  span_types.json:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/span_types.json
```
Comment (trentm) [inline .ci/updatecli.d/update-json-specs.yml:37]: It sure would be nice if updatecli supported a `kind: git-repo-subdir` source or something like that. :)

**Review by trentm — APPROVED** [review]

File: .ci/updatecli.d/update-json-specs.yml (hunk 2 above)
Comment (reakaleek) [reply .ci/updatecli.d/update-json-specs.yml:37]: That would be nice, yeah. 😅

Comment (v1v) [reply .ci/updatecli.d/update-json-specs.yml:37]: https://github.com/updatecli/updatecli/issues/1218 is the feature request I just created, please feel free to add further details if I missed something

Comment (v1v) [reply .ci/updatecli.d/update-json-specs.yml:37]: I've received some feedback about how to implement this particular feature using the existing capabilities, https://github.com/updatecli/updatecli/issues/1218#issuecomment-1463764985

what do you think?

Comment (reakaleek) [reply .ci/updatecli.d/update-json-specs.yml:37]: Looking at the suggested options, there seems to be no way around specifying every single file.

**Pull request merged**